src/structure/pdb.jl

```diff
@@ -30,26 +39,231 @@ end
+By default downloads the PDB file to current working directory; 
+if the keyword argument `ba_number` is set the biological assembly with that number will be downloaded; 
+if the keyword argument `pdb_dir` is set the PDB file is downloaded to the specified directory; 
+if the keyword argument `obsolete` is set `true`, the PDB file is downloaded into the obsolete directory inside `pdb_dir`;
+if the keyword argument `overwrite` is set `true`, then it will overwrite the PDB file if it exists in the `pdb_dir`;
+"""
+function downloadpdb(pdbid::AbstractString, out_filepath::AbstractString="$pdbid.pdb"; pdb_dir::AbstractString=pwd(), obsolete::Bool=false, overwrite::Bool=false, ba_number::Integer=0)
     # Check PDB ID is 4 characters long and only consits of alphanumeric characters
     if length(pdbid) != 4 || ismatch(r"[^a-zA-Z0-9]", pdbid)
         throw(ArgumentError("Not a valid PDB ID: \"$pdbid\""))
+    end 
+    # Check if the PDB file is marked as obsolete
+    if obsolete
+        # Set the download path to obsolete directory inside the "pdb_dir"
+        pdb_dir = joinpath(pdb_dir,"obsolete")
+    end
+    # Check and create directory if it does not exists in filesystem
+    if !isdir(pdb_dir) 
+        info("Creating directory : $pdb_dir")
+        mkpath(pdb_dir)
     end
-    if ba_number == 0
-        download("http://www.rcsb.org/pdb/files/$pdbid.pdb", out_filepath)
+    # Download the PDB file only if it does not exist in the "pdb_dir" and when "overwrite" is false
+    if ispath(joinpath(pdb_dir,out_filepath)) && !overwrite
+        info("PDB Exists : $pdbid")
     else
-        # Will download error page if ba_number is too high
-        download("http://www.rcsb.org/pdb/files/$pdbid.pdb$ba_number", out_filepath)
+        # Temporarily change directory to "pdb_dir" to download the PDB file and revert back to the current working directory 
+        working_dir = pwd()
+        cd(pdb_dir)
+        info("Downloading PDB : $pdbid")
+        if ba_number == 0
+            download("http://www.rcsb.org/pdb/files/$pdbid.pdb", out_filepath)
+        else
+            # Will download error page if ba_number is too high
+            download("http://www.rcsb.org/pdb/files/$pdbid.pdb$ba_number", out_filepath)
+        end
+        cd(working_dir)
+    end           
+end
+
+
+"""
+Downloads a list of PDB files from the RCSB PDB server. 
+By default downloads the PDB files to current working directory; 
+if the keyword argument `pdb_dir` is set the PDB files are downloaded to the specified directory;
+if the keyword argument `obsolete` is set `true`, the PDB files are downloaded into the obsolete directory inside `pdb_dir`;
+if the keyword argument `overwrite` is set `true`, then it will overwrite the PDB file if it exists in the `pdb_dir`;
+"""
+function downloadmultiplepdb(pdbidlist::AbstractArray{String,1}; pdb_dir::AbstractString=pwd(), obsolete::Bool=false, overwrite::Bool=false)
```

Comment:
This name seems a bit long. We could even make this another method of `downloadpdb`, so that function can take either a string or a list.

I think this should be `pdbidlist::Array{String,1}` for the first argument.